gis/gdal/Portfile

```diff
@@ -36,7 +36,13 @@ use_xz              yes
 # error: call to 'abs' is ambiguous
 compiler.blacklist-append {clang < 900}
 
-compiler.cxx_standard 2014
+# Enable use of std::shared_mutex on systems that support it (macOS 10.12+),
+# next major GDAL version (>3.8) will require C++17.
+if {${os.platform} eq "darwin" && ${os.major} < 16} {
+    compiler.cxx_standard 2014
+} else {
+    compiler.cxx_standard 2017
+}
```

Comment:
why can you not set the CXX to 2017 on all systems? It should in all cases select a compiler that support the "shared mutex" thing; either by an Apple provided Clang or a MacPorts installed one.

Reply:
the c++17 and newer features were gradually introduced to libc++.dylb.

so only the /usr/lib/libc++.dylib in 10.12 and newer have it.

```
define _LIBCPP_AVAILABILITY_SHARED_MUTEX                                    \
         __attribute__((availability(macosx,strict,introduced=10.12)))           \
```

Reply:
Thanks @kencu, that confirms what I deducted from the build bots.

@reneeotten GDAL still only requires C++11 (don't know why the Portfile bumped to C++14 with the deletion of the C++11 PG, perhaps because of some dependency, never mind). Only the next major GDAL release (probably 4.0) will require C++17 (the cmake change is already in master). The developers decided to optionally insert C++17 feats, like std::shared_mutex, the included patch improves the detection of its availability.
To sum up: there is **now** no compelling reason to make this C++17 all over and in the process perhaps run into new unexpected issues...